thread_sync.c

```diff
@@ -535,8 +541,7 @@ rb_mutex_sleep(VALUE self, VALUE timeout)
 
     VALUE scheduler = rb_scheduler_current();
     if (scheduler != Qnil) {
-        rb_scheduler_kernel_sleep(scheduler, timeout);
-        mutex_lock_uninterruptible(self);
+        rb_ensure(call_rb_scheduler_block_mutex_sleep, timeout, mutex_lock_uninterruptible, self);
```

Comment:
This ensure seems important and follows the pattern below.
Whatever `call_rb_scheduler_block_mutex_sleep()` does, we need to reacquire the Mutex when getting out of `Mutex#sleep`.

Reply:
I think we should make a separate PR to address this if still necessary.

Reply:
I'll let you integrate this fix since you rejected this PR.

Reply:
Ok.